modules/siddhi-core/src/main/java/io/siddhi/core/query/input/ProcessStreamReceiver.java

```diff
@@ -75,7 +76,8 @@ private void process(ComplexEventChunk<StreamEvent> streamEventChunk) {
         }
         try {
             LatencyTracker latencyTracker = siddhiQueryContext.getLatencyTracker();
-            if (siddhiQueryContext.getSiddhiAppContext().isStatsEnabled() && latencyTracker != null) {
+            if (Level.DETAIL.compareTo(siddhiQueryContext.getSiddhiAppContext().getRootMetricsLevel()) <= 0 &&
```

Comment:
Since the checks are done on the main execution flow this can introduce performance issues. 
I'll do a pert test before merging this.